### PR TITLE
Bounding box

### PR DESCRIPTION
**Pull request by cquiroz**
This is a rework of #502 where we can create a `BoundingBox` as a `ShapeExpression`
The box is now defined by two corner offsets and a function to calculate the widest side (as an angle) is included

Also a few tests are added.

The image below shows the box on the gmos shapes

image

### CODE REVIEW

File: modules/core/shared/src/main/scala/lucuma/core/geom/BoundingOffsets.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+   * {{{
+   * p1 p2
+   * |---|
+   * |   |
+   * |---|
+   * p3 p4
+   * }}}
+   */
+  def vertices: (Offset, Offset, Offset, Offset) =
+    (topLeft, Offset(bottomRight.p, topLeft.q), bottomRight, Offset(topLeft.p, bottomRight.q))
```
Comment (swalker2m) [inline modules/core/shared/src/main/scala/lucuma/core/geom/BoundingOffsets.scala:38]: I think maybe the order here doesn't match the comment.  It looks like `(p1, p2, p4, p3)`?

**Review by swalker2m — COMMENTED** [review]
This looks good.  For context, what do we plan to do with this?

File: modules/core/shared/src/main/scala/lucuma/core/geom/BoundingOffsets.scala (hunk 1 above)
Comment (cquiroz) [reply modules/core/shared/src/main/scala/lucuma/core/geom/BoundingOffsets.scala:38]: Thanks, the comment is wrong. I wanted to keep the vertices on clockwise order

**Comment by cquiroz** [comment]
I'm going to use this to create adql queries with a given geometry. e.g. I can calculate the radius of a shape and do a query there. Eventually more complex geometries would be possible

**Pull request merged**